### PR TITLE
fix(sql): #MA-909 add security to SQL querries in IncidentsTypeService

### PR DESCRIPTION
**Pull request by BastienBenard**

### CODE REVIEW

File: incidents/src/main/java/fr/openent/incidents/service/impl/DefaultIncidentsTypeService.java [hunk 1]
```diff
@@ -88,8 +91,9 @@ public void put(JsonObject incidentTypeBody, Handler<Either<String, JsonObject>>
 
     @Override
     public void delete(Integer incidentTypeId, Handler<Either<String, JsonObject>> handler) {
-        String query = "DELETE FROM  " + Incidents.dbSchema + ".incident_type WHERE id = " + incidentTypeId +
-                " RETURNING id as id_deleted";
+        String query = "DELETE FROM  " + Incidents.dbSchema + ".incident_type WHERE id = ? RETURNING id as id_deleted";
+        JsonArray params = new JsonArray();
+        params.add(incidentTypeId);
         Sql.getInstance().raw(query, SqlResult.validUniqueResultHandler(handler));
```
Comment (qpic-cgi) [inline incidents/src/main/java/fr/openent/incidents/service/impl/DefaultIncidentsTypeService.java]: raw => prepared

File: incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
+package fr.openent.incidents.service.impl;
+
+import fr.openent.incidents.service.IncidentsService;
+import fr.openent.incidents.service.IncidentsTypeService;
+import fr.openent.presences.core.constants.Field;
+import fr.openent.presences.db.DB;
+import fr.openent.presences.db.DBService;
```
Comment (JinDouang) [inline incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java]: pk DBService ?

File: incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+import org.powermock.reflect.Whitebox;
+
+import java.util.Collections;
+
+@RunWith(VertxUnitRunner.class)
+public class IncidentsTypeServiceTest extends DBService {
```
Comment (JinDouang) [inline incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java]: pk DBService ?

File: incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", (Handler) e -> {
+
```
Comment (qpic-cgi) [inline incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java]: Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", null)

File: incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "create", incidentTypeBody, (Handler) e -> {
+
```
Comment (qpic-cgi) [inline incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java]: Whitebox.invokeMethod(..., null)

File: incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "put", incidentTypeBody, (Handler) e -> {
+
```
Comment (qpic-cgi) [inline incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java]: Whitebox.invokeMethod(..., null)

File: incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "delete", 3, (Handler) e -> {
+
```
Comment (qpic-cgi) [inline incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java]: Whitebox.invokeMethod(..., null)

**Review by JinDouang — APPROVED** [review]

**Pull request merged**